tests/complexity.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+var krabbyTest = require('./_baseTest');
+var escomplex = require('escomplex-js');
+var _ = require('lodash');
+var Promise = require('bluebird');
+var fs = Promise.promisifyAll(require("fs"));
+
+var baseConfig = {
+  testConfig: {
+  },
+  files: ['./**/*.js', '!./node_modules/**/*.js', '!./**/node_modules/**/*.js']
+};
+
+var ComplexityTest = new krabbyTest(baseConfig);
+
+ComplexityTest.prototype.analyze = function(files) {
+  var self = this;
+
+  return Promise.map(files, function(fileName) {
+    return fs.readFileAsync(fileName)
+      .then(function(data) {
+        var result = escomplex.analyse(data.toString(), self.config.testConfig);
+        // TODO: Do I even need to do this if the result is already passed all the way to the reporters
+        self.logs.logs.push(result);
+        return result;
+      })
+      .catch(function(e) {
+        // TODO: Why does esprima choke on ./index.js? It might have to do with `#!/usr/bin/env node`
+        // TODO: Where do these get logged?
```

Comment:
Actually, since we pass results through to the reporters do we even need that object? We could probably just require a standard results object and do away with this all together.